test/spec/old_select.wast

```diff
@@ -38,24 +35,6 @@
    (select (unreachable) (i32.const 0) (local.get $cond)))
   (func (export "select_trap_r") (param $cond i32) (result i32)
    (select (i32.const 0) (unreachable) (local.get $cond)))
-
-  (func (export "join-nullref") (param i32) (result externref)
-    (select (result externref) (ref.null) (ref.null) (local.get 0))
-  )
-  (func (export "join-funcref") (param i32) (result externref)
-    (select (result externref)
-      (ref.func $dummy)
-      (ref.null)
-      (local.get 0)
-    )
-  )
-  (func (export "join-externref") (param i32) (param externref) (result externref)
-    (select (result externref)
-      (ref.func $dummy)
-      (local.get 1)
-      (local.get 0)
-    )
-  )
```

Comment:
It would be good to comment these out with a comment that the types should be updated once we have subtyping again to avoid accidentally losing test coverage.

Reply:
Similar situation to https://github.com/WebAssembly/binaryen/pull/3084#discussion_r484212713, where removal of all the subtypes of the tested types make these tests somewhat useless now. Given that there are already other reference types tests for `select` in this test and in `reference-types.wast`, I figured that removing them would probably be fine.